webapp/channels/src/components/file_preview_modal/file_preview_modal.tsx

```diff
@@ -269,7 +292,17 @@ export default class FilePreviewModal extends React.PureComponent<Props, State>
         }
 
         const fileInfo = this.props.fileInfos[this.state.imageIndex];
-        const fileType = Utils.getFileType(fileInfo.extension);
+
+        // Determine file type based on the object type
+        let fileType;
+        if (isFileInfo(fileInfo)) {
+            fileType = Utils.getFileType(fileInfo.extension);
+        } else if (isLinkInfo(fileInfo)) {
+            // For LinkInfo, pass the full URL to getFileType to leverage enhanced URL detection
+            fileType = Utils.getFileType(fileInfo.link);
+        } else {
+            fileType = Utils.getFileType('');
+        }
```

Comment:
Given this duplicates the block above, would it make sense to add a `getFileTypeFromFileInfo` helper method, even if it's just local to this component?

Reply:
100% agree. DRY helper function implemented.